cmd/clusterctl/client/cluster/mover.go

```diff
@@ -42,10 +42,13 @@ import (
 	"sigs.k8s.io/cluster-api/util/yaml"
 )
 
+// ResourceMutatorFunc holds the type for mutators to be applied on resources during a move operation.
+type ResourceMutatorFunc func(u *unstructured.Unstructured)
+
 // ObjectMover defines methods for moving Cluster API objects to another management cluster.
 type ObjectMover interface {
 	// Move moves all the Cluster API objects existing in a namespace (or from all the namespaces if empty) to a target management cluster.
-	Move(namespace string, toCluster Client, dryRun bool) error
+	Move(namespace string, mutators []ResourceMutatorFunc, toCluster Client, dryRun bool) error
```

Comment:
Do we want to make `Move` variadic? The change would be backward compatible with existing callers, because they do not pass any `mutators`.
  
```suggestion
	Move(namespace string, toCluster Client, dryRun bool, mutators ...ResourceMutatorFunc) error
```

Reply:
:+1: